src/inference/src/ie_core.cpp

```diff
@@ -533,7 +533,8 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
             if (batch_mode != config.end()) {
                 const auto disabled = batch_mode->second == CONFIG_VALUE(NO);
                 // no need for this config key in the rest of loading
-                config.erase(batch_mode);
+                if (deviceName.find("AUTO") == std::string::npos && deviceName.find("MULTI") == std::string::npos)
+                    config.erase(batch_mode);
                 if (disabled)
```

Comment:
@myshevts @wangleis please help review if this change is ok. main idea is to not delete this key from virtual device like AUTO/MULTI.

Reply:
Not sure I understand the logic.. presumably we shall check that if user has disabled the Auto-Batching, then we don't need to query the OPTIMAL_BATCH_SIZE to deduce/report the OPTIMAL_NUM_REQUESTS.

If you remove the config, then the ie_core will trigger the Auto-Batching when the AUTO will select the GPU and call a LoadNetwork for that...
Seemngly, the config should be always passed as is (down to the Core) along with the selected device

Reply:
no, I'm not remove the config(which is the original logic, remove the KEY_ALLOW_AUTO_BATCHING after checking its value), I'm preserve the key for -d Auto case.  then AUTO will pass the config as it is to whatever device it chose for loadnetwork.

Reply:
This w/a solution on 2022.1 is ok for me.

ALLOW_AUTO_BATCHING will pass to AUTO as user config during application load network to AUTO. And ALLOW_AUTO_BATCHING will be erased during AUTO load network to HW device like GPU.

Reply:
@songbell  oh, I see this is the code in the ie_core...then it is fine